charts/loki-distributed/templates/runtime-configmap.yaml

```
@@ -0,0 +1,9 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ include "loki.fullname" . }}-runtime
+  labels:
+    {{- include "loki.labels" . | nindent 4 }}
+data:
+  runtime.yaml: |
+    {{ tpl (toYaml .Values.runtimeConfig) . | nindent 4 }}
```

This one will render a leftover `{}` due to given runtimeConfig in values.yaml. If that is not intended, I suggest to change to:
```suggestion
data:
  runtime.yaml: |
    {{- with .Values.runtimeConfig }}
    {{- tpl (toYaml .) . | nindent 4 }}
    {{- end }}
```

---

It was not really intended, but after testing it, it appears to work :) Anyway, i'm totally agree with you, an empty file is preferable, even if mimir-distributed doesn't burden itself with such considerations :) (https://github.com/grafana/mimir/blob/main/operations/helm/charts/mimir-distributed/templates/runtime-configmap.yaml#L10)

---

Ooops, emergency rollback, loki does not appreciate empty yaml files :)
```
level=info ts=2022-09-21T13:54:10.175105196Z caller=main.go:103 msg="Starting Loki" version="(version=2.6.1, branch=HEAD, revision=6bd05c9a4)"
level=info ts=2022-09-21T13:54:10.175611823Z caller=server.go:288 http=[::]:3100 grpc=[::]:9095 msg="server listening on addresses"
level=info ts=2022-09-21T13:54:10.181660119Z caller=module_service.go:82 msg=initialising module=usage-report
level=info ts=2022-09-21T13:54:10.18263276Z caller=module_service.go:82 msg=initialising module=runtime-config
level=error ts=2022-09-21T13:54:10.183541249Z caller=loki.go:386 msg="module failed" module=runtime-config error="invalid service state: Failed, expected: Running, failure: failed to load runtime config: load file: EOF"
level=info ts=2022-09-21T13:54:10.182623112Z caller=module_service.go:82 msg=initialising module=server
level=error ts=2022-09-21T13:54:10.18427708Z caller=loki.go:386 msg="module failed" module=server error="context canceled"
level=error ts=2022-09-21T13:54:10.185574901Z caller=loki.go:386 msg="module failed" module=compactor error="failed to start compactor, because it depends on module runtime-config, which has failed: invalid service state: Failed, expected: Running, failure: invalid service state: Failed, expected: Running, failure: failed to load runtime config: load file: EOF"
level=error ts=2022-09-21T13:54:10.185679194Z caller=loki.go:386 msg="module failed" module=memberlist-kv error="failed to start memberlist-kv, because it depends on module server, which has failed: context canceled"
level=info ts=2022-09-21T13:54:10.185989439Z caller=module_service.go:114 msg="module stopped" module=usage-report
level=info ts=2022-09-21T13:54:10.186188339Z caller=loki.go:375 msg="Loki stopped"
failed services
github.com/grafana/loki/pkg/loki.(*Loki).Run
	/src/loki/pkg/loki/loki.go:421
main.main
	/src/loki/cmd/loki/main.go:105
runtime.main
	/usr/local/go/src/runtime/proc.go:255
runtime.goexit
	/usr/local/go/src/runtime/asm_amd64.s:1581
level=error ts=2022-09-21T13:54:10.18640974Z caller=log.go:103 msg="error running loki" err="failed services\ngithub.com/grafana/loki/pkg/loki.(*Loki).Run\n\t/src/loki/pkg/loki/loki.go:421\nmain.main\n\t/src/loki/cmd/loki/main.go:105\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:255\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1581"
```

---

How about a yaml file with `---`?

```suggestion
data:
  runtime.yaml: |
    {{- with .Values.runtimeConfig }}
    {{- tpl (toYaml .) . | nindent 4 }}
    {{- else }}
    ---
    {{- end }}
```